apps/infrastructure/src/cloudfront.tf

```
@@ -209,3 +209,53 @@ resource "aws_cloudfront_distribution" "storybook" {
     minimum_protocol_version       = "TLSv1.2_2021"
   }
 }
+
+## MediaLibrary Strapi Multitenant CDN
+resource "aws_cloudfront_distribution" "cms_multitenant_medialibrary" {
+  for_each = {
+    for key, config in var.websites_configs :
+    key => config
+  }
+  origin {
+    domain_name = aws_s3_bucket.cms_multitenant_medialibrary_bucket[each.key].bucket_regional_domain_name
+    origin_id   = aws_s3_bucket.cms_multitenant_medialibrary_bucket[each.key].bucket
+
+    s3_origin_config {
+      origin_access_identity = aws_cloudfront_origin_access_identity.main.cloudfront_access_identity_path
+    }
+  }
+
+  enabled         = true # enable CloudFront distribution
+  is_ipv6_enabled = true
+  comment         = "CloudFront distribution for the medialibrary cms."
```

If you agree, we can add the name of the tenant in the comment
```suggestion
  comment         = "CloudFront distribution for the ${each.key} medialibrary cms."
```